tests/src/Unit/Plugin/Notifier/SmsTest.php

```diff
@@ -0,0 +1,289 @@
+   */
+  protected $pluginDefinition;
+
+  /**
+   * Mocked logger.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * Mocked entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Mocked renderer.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * Mocked phone number provider.
+   *
+   * @var \Drupal\sms\Provider\PhoneNumberProviderInterface
+   */
+  protected $phoneNumberProvider;
+
+  /**
+   * Test setup.
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // SmsMessage.php#L326 is calling uuid service directly. Need to setup
+    // fake container for this test.
+    $container = new ContainerBuilder();
+    $container->set('uuid', $this->getMockUuidService());
+    \Drupal::setContainer($container);
+
+    $this->configuration = [];
+    $this->pluginId = $this->randomMachineName();
+    $this->pluginDefinition = [
+      'title' => $this->randomMachineName(),
+      'viewModes' => [
+        'sms_message',
+      ],
+    ];
+    $this->logger = $this->prophesize(LoggerChannelInterface::class)->reveal();
+    $this->entityTypeManager = $this->prophesize(EntityTypeManagerInterface::class)->reveal();
+    $this->renderer = $this->prophesize(RendererInterface::class)->reveal();
+    $this->phoneNumberProvider = $this->prophesize(PhoneNumberProviderInterface::class)->reveal();
+  }
+
+  /**
+   * Test Sms::deliver happy path.
+   *
+   * @covers \Drupal\message_notify\Plugin\Notifier\Sms::deliver
+   */
+  public function testDeliver() {
+    $this->phoneNumberProvider = $this->getMockRespondingPhoneProvider(NULL);
+
+    $user = $this->getMockUser();
+    $message = $this->getMockMessage($user);
+    $notifier = $this->getNotifier($message);
+
+    $output = ['sms_message' => 'Hello, World!'];
+    $this->assertTrue($notifier->deliver($output));
+  }
+
+  /**
+   * Test Sms::deliver empty message.
+   *
+   * @covers \Drupal\message_notify\Plugin\Notifier\Sms::deliver
+   */
+  public function testDeliverEmptyMessage() {
+    $this->phoneNumberProvider = $this->getMockRespondingPhoneProvider(NULL);
+
+    $user = $this->getMockUser();
+    $message = $this->getMockMessage($user);
+    $notifier = $this->getNotifier($message);
+
+    $output = ['sms_message' => ''];
+    $this->assertFalse($notifier->deliver($output));
+  }
+
+  /**
+   * Test Sms::deliver with delivery failure.
+   *
+   * @covers \Drupal\message_notify\Plugin\Notifier\Sms::deliver
+   */
+  public function testDeliverErrorCode() {
+    $this->phoneNumberProvider = $this->getMockRespondingPhoneProvider(SmsMessageResultStatus::NO_CREDIT);
+
+    $user = $this->getMockUser();
+    $message = $this->getMockMessage($user);
+    $notifier = $this->getNotifier($message);
+
+    $output = ['sms_message' => 'Hello, World!'];
+    $this->assertFalse($notifier->deliver($output));
```

Comment:
Maybe we should also check the logger for the correct message? (IMO not a must, but a nice to have)